x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/EsField.java

```diff
@@ -85,6 +87,15 @@ public boolean isExact() {
         return true;
     }
 
+    /**
+     * {True, null} if this field has an exact field which can be retrieved by {@link EsField#getExactField()}
+     * {False, String} if this field doesn't have an underlying exact field, the v2 of the tuple is a string containing
+     * an error message explaining why that is.
+     */
+    public Tuple<Boolean, String> hasExact() {
```

Comment:
I think having both `isExact` and `hasExact` is confusing - is the former still used anywhere? I opt for refactoring isExact to return the tuple instead (which can be promoted to a dedicated inner class such (`Exact`) to make its use easier and its intent clearer).

Reply:
I agree with the confusing `isExact` but it's used externally here: https://github.com/elastic/elasticsearch/pull/39427/files#diff-1059b5208ef7d10a0fdc89b5dad914edR90 and has slightly different semantics than `hasExact`.

`hasExact` returns <TRUE, null> if it's exact, or has an exact underlying field, so we don't know which is the case.

But if we introduce a class `Exact` to replace the the `Tuple` we can include this info there and remove the confusing `isExact()` method. Thx!